src/client/components/stage-details/stage-details.component.js

```diff
@@ -0,0 +1,171 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import './stage-details.style.css';
+import SectionHeader from '../section-header/section-header.component';
+import Button from '../button/button.component';
+
+function capitalizeString(str) {
+  return str[0].toUpperCase().concat(str.slice(1, str.length));
+}
+
+export default function StageDetails({
+  cropID,
+  parametersAndLabels,
+  stages,
+  levels,
+  data,
+}) {
+  // Get the parameter data for crop with id of cropID
+  const [cropData, setCropData] = useState([]);
+
+  useEffect(() => {
+    setCropData(
+      data.filter((dataPoint) => parseInt(dataPoint.fk_crop_id, 10) === cropID),
+    );
+  }, [cropID, data]);
+
+  // Select the last stage to display as default
+  const [selectedStageNumber, setSelectedStageNumber] = useState(
+    Math.max(...stages.map((s) => s.id)),
+  );
+
+  // Get data for the currently selected stage
+  const [dataForStage, setDataForStage] = useState([]);
+
+  useEffect(() => {
+    setDataForStage(
+      cropData
+        ? cropData.filter(
+            (dataPoint) =>
+              parseInt(dataPoint.fk_crop_stage_id, 10) === selectedStageNumber,
+          )
+        : [],
+    );
+  }, [selectedStageNumber, cropData]);
+
+  function handleInputChange(e, level, parameter) {
+    const valueEntered = e.target.value;
+    e.preventDefault();
+    setCropData((prev) => {
+      return prev.map((dataPoint) => {
+        if (
+          Number(dataPoint.fk_crop_stage_id) === selectedStageNumber &&
+          dataPoint.parameter === parameter
+        ) {
+          const updatedDataPoint = {
+            ...dataPoint,
+            [`${level}_value`]: valueEntered,
+          };
+          return updatedDataPoint;
+        }
+        return dataPoint;
+      });
+    });
+  }
+
+  function saveUpdatedCropData(e) {
+    e.preventDefault();
+    /* TO DO: SAVE UPDATED cropData */
+  }
```

Comment:
Yes, I added an `onChange` prop to the component which it will be receiving from the container. 

Now it's 
```
  function saveUpdatedCropData(e) {
    e.preventDefault();
    onChange(cropData);
  }
```